systempoweraction/src/main/java/com/stream_pi/systempoweraction/SystemPowerAction.java

```diff
@@ -0,0 +1,163 @@
+package com.stream_pi.systempoweraction;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+
+import com.stream_pi.action_api.actionproperty.property.ListValue;
+import com.stream_pi.action_api.actionproperty.property.Property;
+import com.stream_pi.action_api.actionproperty.property.Type;
+import com.stream_pi.action_api.externalplugin.NormalAction;
+import com.stream_pi.util.platform.Platform;
+import com.stream_pi.util.exception.MinorException;
+import com.stream_pi.util.version.Version;
+
+import java.io.IOException;
+import java.io.OutputStream;
+
+public class SystemPowerAction extends NormalAction
+{
+    public SystemPowerAction()
+    {
+        setName("Power Action");
+        setAuthor("quimodotcom");
+        setHelpLink("https://github.com/quimodotcom/essential-actions");
+        setVersion(new Version(1,0,0));
+        setCategory("System");
+        
+        states = new ArrayList<>();
+        states.addAll(Arrays.asList(
+                new ListValue("Shutdown"),
+                new ListValue("Restart"),
+                new ListValue("Sleep")
+        ));
+    }
+    
+    private final ArrayList<ListValue> states;
+    
+    @Override
+    public void initProperties() throws MinorException
+    {
+        Property status = new Property("state", Type.LIST);
+        status.setListValue(states);
+        status.setDisplayName("Power State");
+
+        addClientProperties(status);
+    }
+    
+    @Override
+    public void onActionClicked() throws MinorException
+    {
+        Platform plat = getPlatform();
+        String state = states.get(getClientProperties().getSingleProperty("state").getSelectedIndex()).getName().toString();
+        switch (state)
+        {
+            case "Shutdown":
+                Shutdown(plat);
+                break;
+            
+            case "Restart":
+                Restart(plat);
+                break;
+                
+            case "Sleep":
+                Sleep(plat);
+                break;     
+        }
+    }
+    
+    public void Shutdown(Platform plat) throws MinorException
+    {
+        String shutdownCommand = new String();
+        try{
+            if (null != plat) switch (plat) {
+            case MAC:
+                Process p = Runtime.getRuntime().exec
```

Comment:
What is the difference between 
```
Runtime.getRuntime().exec("osascript -e 'tell application \"System Events\"' -e \"shut down\" -e 'end tell'");
```
and this?

Reply:
fixed

Reply:
so it writes a file to the mac dir and enables the mac system permissions to the server